core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1878,6 +1906,21 @@ public void onCustomAction(String action, Bundle extra) {
                 onFastForward();
             } else if (CUSTOM_ACTION_REWIND.equals(action)) {
                 onRewind();
+            } else if (CUSTOM_ACTION_SKIP_TO_NEXT.equals(action)) {
+                onSkipToNext();
+            } else if (CUSTOM_ACTION_CHANGE_PLAYBACK_SPEED.equals(action)) {
+                List<Float> selectedSpeeds = UserPreferences.getPlaybackSpeedArray();
+                int speedPosition = selectedSpeeds.indexOf(mediaPlayer.getPlaybackSpeed());
+
+                float newSpeed;
+                if (speedPosition == selectedSpeeds.size() - 1) {
+                    // This is the last element. Wrap instead of going over the size of the list.
+                    newSpeed = selectedSpeeds.get(0);
```

Comment:
This will crash if there are no speed presets saved.

Reply:
Yep, that will - I'll fix it.